File: 백준/Silver/10828. 스택/스택.py
import sys
input = sys.stdin.readline

s = []

n = int(input())

for _ in range(n):
    cmd = input().split()
    
    if cmd[0] == "push":
        s.append(cmd[1])
    elif cmd[0] == "top":
        if len(s) == 0 :
            print(-1)
        else :
            print(s[-1])
    elif cmd[0] == "size":
        print(len(s))
    elif cmd[0] == "pop":
        if len(s) == 0 :
            print(-1)
        else :
            print(s.pop())
    elif cmd[0] == "empty":
        if len(s) == 0 :
            print(1)
        else :
            print(0)

    
    

File: 백준/Silver/10828. 스택/스택.java


import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        Stack<Integer> stack = new Stack<>();
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        // 1. 입력
        int N = Integer.parseInt(br.readLine());

        for (int i = 0; i < N; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine(), " ");
            String command = st.nextToken();

            if (command.equals("push") ) {
                int n = Integer.parseInt(st.nextToken());
                stack.push(n);
            } else if (command.equals("pop")) {
                if(stack.isEmpty()) sb.append("-1").append("\n");
                else sb.append(stack.pop()).append("\n");
            } else if (command.equals("size")) {
                sb.append(stack.size()).append("\n");
            } else if (command.equals("empty")) {
                if(stack.isEmpty()) sb.append("1").append("\n");
                else sb.append("0").append("\n");
            } else if (command.equals("top")) {
                if(stack.isEmpty()) sb.append("-1").append("\n");
                else sb.append(stack.peek()).append("\n");
            }
        }

        System.out.println(sb);

        // push면 분리해서 숫자 넣기
        //

    }
}
